tests/unit/functions/test_functions_date_functions.py

```diff
@@ -31,162 +33,163 @@ def test_ticks(date_time_str: str, expected_ticks: int) -> None:
+    result: str = date_functions.add_to_time("2023-11-24T12:11:49.7531321Z", interval, time_unit)
     assert result == expected
 
 
 def test_add_to_time_unknown_interval_throws_error() -> None:
     with pytest.raises(ValueError):
-        date_functions.add_to_time("2023-11-24T12:11:49.753132Z", 5, "Unknown")
+        date_functions.add_to_time("2023-11-24T12:11:49.7531321Z", 5, "Unknown")
 
 
 @pytest.mark.parametrize(
     "interval, time_unit, expected",
     [
-        param(-1, "Second", "2023-11-24T12:11:48.753132Z", id="negative-second"),
-        param(-1, "Minute", "2023-11-24T12:10:49.753132Z", id="negative-minute"),
-        param(-1, "Hour", "2023-11-24T11:11:49.753132Z", id="negative-hour"),
-        param(-1, "Day", "2023-11-23T12:11:49.753132Z", id="negative-day"),
-        param(-1, "Week", "2023-11-17T12:11:49.753132Z", id="negative-week"),
-        param(-1, "Month", "2023-10-24T12:11:49.753132Z", id="negative-month"),
-        param(-1, "Year", "2022-11-24T12:11:49.753132Z", id="negative-year"),
-        param(5, "Second", "2023-11-24T12:11:54.753132Z", id="positive-second"),
-        param(5, "Minute", "2023-11-24T12:16:49.753132Z", id="positive-minute"),
-        param(5, "Hour", "2023-11-24T17:11:49.753132Z", id="positive-hour"),
-        param(5, "Day", "2023-11-29T12:11:49.753132Z", id="positive-day"),
-        param(5, "Week", "2023-12-29T12:11:49.753132Z", id="positive-week"),
-        param(5, "Month", "2024-04-24T12:11:49.753132Z", id="positive-month"),
-        param(5, "Year", "2028-11-24T12:11:49.753132Z", id="positive-year"),
+        param(-1, "Second", "2023-11-24T12:11:48.7531321Z", id="negative-second"),
+        param(-1, "Minute", "2023-11-24T12:10:49.7531321Z", id="negative-minute"),
+        param(-1, "Hour", "2023-11-24T11:11:49.7531321Z", id="negative-hour"),
+        param(-1, "Day", "2023-11-23T12:11:49.7531321Z", id="negative-day"),
+        param(-1, "Week", "2023-11-17T12:11:49.7531321Z", id="negative-week"),
+        param(-1, "Month", "2023-10-24T12:11:49.7531321Z", id="negative-month"),
+        param(-1, "Year", "2022-11-24T12:11:49.7531321Z", id="negative-year"),
+        param(5, "Second", "2023-11-24T12:11:54.7531321Z", id="positive-second"),
+        param(5, "Minute", "2023-11-24T12:16:49.7531321Z", id="positive-minute"),
+        param(5, "Hour", "2023-11-24T17:11:49.7531321Z", id="positive-hour"),
+        param(5, "Day", "2023-11-29T12:11:49.7531321Z", id="positive-day"),
+        param(5, "Week", "2023-12-29T12:11:49.7531321Z", id="positive-week"),
+        param(5, "Month", "2024-04-24T12:11:49.7531321Z", id="positive-month"),
+        param(5, "Year", "2028-11-24T12:11:49.7531321Z", id="positive-year"),
     ],
 )
-@freeze_time("2023-11-24T12:11:49.753132Z")
 def test_get_future_time(interval: int, time_unit: str, expected: str) -> None:
+    CSharpDateTime.utcnow = lambda: CSharpDateTime.parse("2023-11-24T12:11:49.7531321Z")
```

Comment:
Reminder: see `Monekypatch` of pytest above (won't add another reminder for other cases).